### PR TITLE
Fix itertools count repr

### PR DESCRIPTION
**Pull request by oow214**

### CODE REVIEW

**Comment by fanninpm** [comment]
Feel free to ignore those Clippy failures. @youknowone likely missed them in #3817.

**Comment by fanninpm** [comment]
In `Lib/test/test_itertools.py`, if you comment out the `@unittest.expectedFailure` decorator above `def test_count(self):` (about line 537 or so), what happens? I suspect that the `__reduce__()` method of `itertools.count` is causing the pickle portion of that test to fail.

**Comment by youknowone** [comment]
please do not use `git merge`. would you try to rebase it?

https://docs.gitlab.com/ee/topics/git/git_rebase.html

File: vm/src/stdlib/itertools.rs [hunk 1]
```diff
@@ -220,9 +220,13 @@ mod decl {
         }
 
         #[pymethod(magic)]
-        fn repr(&self) -> PyResult<String> {
-            let cur = self.cur.read();
-
+        fn repr(&self, vm: &VirtualMachine) -> PyResult<String> {
+            let mut cur = format!("{}", self.cur.read().clone().repr(vm)?);
+            let step = format!("{}", self.step.clone());
+            if step != "1" {
```
Comment (youknowone) [inline vm/src/stdlib/itertools.rs:226]: I expected comparing the value instead of comparing the result.
Do you know what's the original way?

Comment (Snowapril) [reply vm/src/stdlib/itertools.rs:226]: cpython is also comparing the value (https://github.com/python/cpython/blob/3.10/Modules/itertoolsmodule.c#L4305-L4310).

File: vm/src/stdlib/itertools.rs [hunk 2]
```diff
@@ -221,13 +221,12 @@ mod decl {
 
         #[pymethod(magic)]
         fn repr(&self, vm: &VirtualMachine) -> PyResult<String> {
-            let mut cur = format!("{}", self.cur.read().clone().repr(vm)?);
-            let step = format!("{}", self.step.clone());
-            if step != "1" {
-                cur.push_str(", ");
-                cur.push_str(&step);
+            let cur = format!("{}", self.cur.read().clone().repr(vm)?);
+            let step = self.step.as_u32_mask().clone();
```
Comment (Snowapril) [inline vm/src/stdlib/itertools.rs:225]:
```suggestion
            let step = self.step.as_bigint();
```

**Comment by Snowapril** [comment]
After #3830 merged, the clippy error also will be passed.

**Review by youknowone — APPROVED** [review]
Thank you for contributing!

**Pull request merged**